packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -517,12 +531,14 @@ class PathNode implements UpPath {
      * the caller must ensure that the reference from child to parent is also removed (or the child is no longer used).
      */
     public removeChild(child: PathNode): void {
-        assert(this.status === Status.Alive, "PathNode must be alive");
+        // It's possible for this node to be `Dead` because we don't remove the children of path nodes
+        // that correspond to deleted content.
+        assert(this.status !== Status.Disposed, "PathNode must not be disposed");
         const key = child.parentField;
         const field = this.children.get(key);
         // TODO: should do more optimized search (ex: binary search or better) using child.parentIndex()
         // Note that this is the index in the list of child paths, not the index within the field
-        const childIndex = field?.indexOf(child);
+        const childIndex = field?.indexOf(child) ?? -1;
         assert(childIndex !== undefined, 0x35c /* child must be parented to be removed */);
```

Comment:
you made this assert impossible with the above change.

I think this code no longer asserts if the child's key does not exist in the children map.

perhaps you intended to unify that with checking for -1 from indexOf?

Reply:
Oops, yeah I changed this part multiple times and ended up reverting the assert line.
Before this change the assert would only be triggered in the case where `field` was `undefined`.